streams/src/test/java/org/apache/kafka/streams/integration/SwallowUnknownTopicErrorIntegrationTest.java

```diff
@@ -140,40 +166,27 @@ private void closeApplication(final Properties streamsConfiguration) throws Exce
     public void shouldThrowStreamsExceptionWithMissingTopicAndDefaultExceptionHandler() throws Exception {
         final Properties streamsConfiguration = getCommonProperties();
         kafkaStreams = new KafkaStreams(topology, streamsConfiguration);
-        kafkaStreams.setUncaughtExceptionHandler(e -> {
-            caughtException.set(e);
-            return StreamsUncaughtExceptionHandler.StreamThreadExceptionResponse.SHUTDOWN_CLIENT;
-        });
         kafkaStreams.start();
         TestUtils.waitForCondition(
             () -> kafkaStreams.state() == State.RUNNING,
             timeoutMs,
             () -> "Kafka Streams application did not reach state RUNNING in " + timeoutMs + " ms"
         );
 
-        TestUtils.waitForCondition(
-            this::receivedUnknownTopicOrPartitionException,
-            timeoutMs,
-            () -> "Did not receive UnknownTopicOrPartitionException"
+        final Properties props = TestUtils.consumerConfig(
+            CLUSTER.bootstrapServers(),
+            "consumer",
+            IntegerDeserializer.class,
+            StringDeserializer.class
         );
 
-        TestUtils.waitForCondition(
-            () -> kafkaStreams.state() == State.ERROR,
-            timeoutMs,
-            () -> "Kafka Streams application did not reach state ERROR in " + timeoutMs + " ms"
+        IntegrationTestUtils.verifyKeyValueTimestamps(
+            props,
+            STREAM_OUTPUT,
+            Collections.singletonList(new KeyValueTimestamp<>(1, "A", CLUSTER.time.milliseconds() + 2))
         );
-        closeApplication(streamsConfiguration);
-    }
 
-    private boolean receivedUnknownTopicOrPartitionException() {
-        if (caughtException.get() == null) {
-            return false;
-        }
-
-        assertInstanceOf(StreamsException.class, caughtException.get());
-        assertInstanceOf(TimeoutException.class, caughtException.get().getCause());
-        assertInstanceOf(UnknownTopicOrPartitionException.class, caughtException.get().getCause().getCause());
-
-        return true;
+        closeApplication(streamsConfiguration);
+        assertEquals(kafkaStreams.state(), State.NOT_RUNNING);
```

Comment:
Just curious - do we need to assert the application is not running at the end of test?  I don't recall us doing this in other integration tests.

Reply:
Not sure yet -- the idea was to check if there was not crash, and final state is not ERROR -- but I think I want to change this test a little bit more anyway... (not 100% happy with the test yet)

Reply:
After testing a few more things, the test is actually ok -- when I change the test handler to return RETRY or FAIL (instead of CONTINUE), the test does on pass. Will remove this check; it's not necessary.